File: calculadora/calculadora_v2.py
# Calculadora em Python
from functools import reduce
from operator import add, sub, mul, truediv


def obter_operacao():
    print("******************* Calculadora em Python *******************")
    print("1. Soma")
    print("2. Subtração")
    print("3. Multiplicação")
    print("4. Divisão")
    print("0. Sair")
    return int(input("Escolha a operação: "))


def obter_numeros():
    lista = []
    contador = 1
    while True:
        numero = int(input(f"Numero {contador}: "))
        lista.append(numero)
        contador += 1
        if contador > 2:
            letra = input("\nMais um número? (S/N): ").upper()
            if letra == "N":
                break
    return lista


def calcular(operacao, numeros):
    operacoes = {1: add, 2: sub, 3: mul, 4: truediv}
    if operacao == 4 and 0 in numeros:
        print("\nNão pode dividir por zero\n")
        return 0
    return reduce(operacoes[operacao], numeros)


def main():
    sair = False
    while not sair:
        operacao = obter_operacao()
        if operacao == 0:
            sair = True
            continue
        if operacao < 0 or operacao > 4:
            print("\nOperação inválida!\n")
            continue
        numeros = obter_numeros()
        print(f"\nNúmeros utilizados: {', '.join(map(str, numeros))}")
        resultado = calcular(operacao, numeros)
        print(f"\nO resultado é {resultado}\n")


if __name__ == "__main__":
    main()

File: calculadora/calculadora_v2.php
<?php
// Calculadora em PHP

enum Operacoes: int{
    case add = 1;
    case sub = 2;
    case mul = 3;
    case div = 4;
}

function reduce(int $operacao, $numeros): float {
    switch($operacao){
        case Operacoes::add->value:
            $resultado = 0;
            break;
        case Operacoes::sub->value:
            $resultado = $numeros[0] * 2;
            break;
        case Operacoes::mul->value:
            $resultado = 1;
            break;
        case Operacoes::div->value:
            $resultado = $numeros[0] ** 2;
    }    
    foreach($numeros as $numero){
        switch($operacao){
            case Operacoes::add->value:
                $resultado += $numero;
                break;
            case Operacoes::sub->value:
                $resultado -= $numero;
                break;
            case Operacoes::mul->value:
                $resultado *= $numero;
                break;
            case Operacoes::div->value:
                $resultado /= $numero;
        }    
    }
    return $resultado;
}

function obter_operacao(): int {
    echo "******************* Calculadora em PHP *******************\n";
    echo "1. Soma\n";
    echo "2. Subtração\n";
    echo "3. Multiplicação\n";
    echo "4. Divisão\n";
    echo "0. Sair\n";
    return (int) readline("Escolha a operação: ");
}

function obter_numeros(): array {
    $lista = [];
    $contador = 1;
    while (true){
        $numero = (int) readline("Numero $contador: ");
        $lista[] = $numero;
        $contador += 1;
        if ($contador > 2){
            $letra = strtoupper(readline("\nMais um número? (S/N): "));
            if ($letra == "N"){
                break;
            }
        }
    }
    return $lista;
}

function calcular($operacao, $numeros): float {
    if ($operacao == 4 && in_array(0,$numeros)){
        echo "\nNão pode dividir por zero\n";
        return 0;
    }
    return reduce($operacao, $numeros);
}

function main(){
    $sair = False;
    while (!$sair){
        $operacao = obter_operacao();
        if ($operacao == 0){
            $sair = True;
            continue;
        }
        if ($operacao < 0 or $operacao > 4){
            echo "\nOperação inválida!\n";
            continue;
        }
        $numeros = obter_numeros();
        echo "\nNúmeros utilizados: " . implode(',',$numeros);
        $resultado = calcular($operacao, $numeros);
        echo "\nO resultado é $resultado\n";
    }
}

main();